otherlibs/stdune/src/int.ml

```diff
@@ -12,7 +12,7 @@ include Comparable.Make (T)
 
 let equal (a : t) b = a = b
 
-let hash (t : t) = t
+let hash (t : t) = Poly.hash t
```

Comment:
Do you know what it does?

For reference, here is an implementation from Base:

```ocaml
    (* This hash was chosen from here: https://gist.github.com/badboy/6267743

       It attempts to fulfill the primary goals of a non-cryptographic hash function:

       - a bit change in the input should change ~1/2 of the output bits
       - the output should be uniformly distributed across the output range
       - inputs that are close to each other shouldn't lead to outputs that are close to
         each other.
       - all bits of the input are used in generating the output

       In our case we also want it to be fast, non-allocating, and inlinable.  *)
    let[@inline always] hash_int (t : int) =
      let t = lnot t + (t lsl 21) in
      let t = t lxor (t lsr 24) in
      let t = t + (t lsl 3) + (t lsl 8) in
      let t = t lxor (t lsr 14) in
      let t = t + (t lsl 2) + (t lsl 4) in
      let t = t lxor (t lsr 28) in
      t + (t lsl 31)
    ;;
```

Reply:
Here is the implementation from the stdlib:

https://github.com/ocaml/ocaml/blob/e3bf4006d2b79fa84750dd3624e7b664ec88f1b2/runtime/hash.c#L56

The implementation in base seems a lot faster (no ffi or tag dispatch overhead). I'm happy to use it instead

Reply:
Yeah, let's use the one from Base.